SEO optimizations for "Pulumi vs Cloud Templates" (#2515)
pulumi/docs: content/docs/intro/vs/cloud_templates.md

aliases: ["/docs/reference/vs/cloud_templates/"]
---

> ## What is CloudFormation?

Reviewer: This heading doesn't work, as the content addresses ARM also. IMO we should just remove it.

Author: 👍

All major cloud providers offer their own form of infrastructure-as-"code" solution, typically by way of JSON or
YAML-based templating solutions. This includes AWS CloudFormation and Azure Resource Manager (ARM) templates.

These markup-based configuration files are often uploaded to a hosted service in the target cloud, where a hosted
service will then process the files to create, update, or delete resources as necessary.

> ## The Pulumi Difference

Reviewer: Same with this one -- the phrase has an advertising-y ring to it and doesn't add much value, and since already seem to rank #1 for "difference between pulumi and cloudformation", presumably because of the "vs" in the title, so I'd suggest dropping this heading, too.

Author: 👍

> Pulumi offers a multi-cloud alternative to more restrictive infrastructure-as-code template solutions, such as AWS CloudFormation. Pulumi allows you to streamline processes through managing resources from different platforms all in one place, using real code, in your favorite language.

Reviewer: ```suggestion
Pulumi offers a multi-cloud alternative to more restrictive infrastructure-as-code template solutions like AWS CloudFormation. Pulumi allows you to streamline processes by managing resources from different platforms all in one place, using real code, in your favorite language.
```

Author: 👍

> Pulumi's model shares a lot with these systems. Although Pulumi programs are written in imperative, familiar languages,

Reviewer: This paragraph doesn't make sense wedged in where it is. It interrupts the flow between the second and fourth paragraphs, and is essentially saying the same thing as the fifth. We should reposition it, make it work, or remove it.

Author: Since it is redundant I opted to remove it. No reason to have redundant content IMO.
